jsonnet/jsonnet.bzl

```diff
@@ -180,15 +180,18 @@ def _jsonnet_to_json_test_impl(ctx):
   diff_command = ""
   if ctx.file.golden:
     golden_files += [ctx.file.golden]
+    # Note that we only run jsonnet to canonicalize the golden output if the
+    # expected return code is 0. Otherwise, the golden file contains the
+    # expected error output.
     if ctx.attr.regex:
       diff_command = _REGEX_DIFF_COMMAND % (
-          ctx.file.jsonnet.short_path,
+          ctx.file.jsonnet.short_path if ctx.attr.error == 0 else 'cat',
```

Comment:
This made the change hard to understand. jssonnet is an  executable right? In that case, I recommend using ctx.executable just to make it clear that you just using a different command.

(also for the other diff command)